sympy/solvers/solveset.md

```diff
@@ -0,0 +1,72 @@
+# Solveset Documentation
+
+## What's wrong with solve():
+
+SymPy already has a pretty powerful `solve` function. But it has a lot of major
+issues
+
+1. It doesn't have a consistent output for various types of solutions
+   It needs to return a lot of types of solutions consistently:
+   * single solution : ` x == 1`
+   * Multiple solutions: `x**2 == 1`
+   * No Solution: `x**2 + 1 == 0; x is real`
+   * Interval of solution: `floor(x) == 0`
+   * Infinitely many solutions: `sin(x) == 0`
+   * Multivariate functions with point solutions `x**2 + y**2 == 0`
+   * Multivariate functions with non point solution `x**2 + y**2 == 1`
+   * System of equations `x + y == 1` and `x - y == 0`
+   * Relational `x > 0`
+   * And the most important case "We don't Know"
```

Comment:
I would put this as a separate point. More generally, there is no distinction between "we know we found all the solutions" and "we only found some solutions, but there could be more". This is particularly evident when no solutions are returned, but it applies in every case.